adapter/prometheus/server.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"io"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// TODO: rethink when we switch to go1.8. the graceful shutdown changes
+// coming to http.Server should help tremendously
+
+type (
+	server interface {
+		io.Closer
+
+		Start() error
+	}
+
+	serverInst struct {
+		server
+
+		addr       string
+		connCloser io.Closer
+	}
+)
+
+const (
+	metricsPath = "/metrics"
+	defaultAddr = ":42422"
+)
+
+func newServer(addr string) server {
+	return &serverInst{addr: addr}
+}
+
+func (s *serverInst) Start() error {
+	var listener net.Listener
+	srv := &http.Server{Addr: s.addr}
+	listener, err := net.Listen("tcp", s.addr)
+	if err != nil {
+		return err
```

Comment:
Updated with custom error message.

I don't think we want to log and return an error.  Wherever the error is handled can log appropriately.